src/webgpu/util/conversion.ts

```diff
@@ -91,7 +91,19 @@ export function float32ToFloatBits(
     return (((1 << exponentBits) - 1) << mantissaBits) | ((1 << mantissaBits) - 1);
   }
 
+  const buf = new DataView(new ArrayBuffer(Float32Array.BYTES_PER_ELEMENT));
+  buf.setFloat32(0, n, true);
+  const bits = buf.getUint32(0, true);
+  // bits (32): seeeeeeeefffffffffffffffffffffff
+
+  // 0 or 1
+  const sign = (bits >> 31) & signBits;
+
   if (n === 0) {
+    if (sign === 1) {
```

Comment:
Yes, this could be shortened to avoid the if.
Could have written  `sign << (exponentBits + mantissaBits)`.
But that's a little more cryptic.

Reply:
I'm in favor of the more explicit code here.